server/src/main/java/org/opensearch/index/translog/transfer/listener/TranslogTransferListener.java

```diff
@@ -32,4 +32,8 @@ public interface TranslogTransferListener {
      * @throws IOException the exception during the transfer of data
      */
     void onUploadFailed(TransferSnapshot transferSnapshot, Exception ex) throws IOException;
+
+    default void close() {
```

Comment:
should this interface extend `AutoCloseable` ?

Reply:
Even if we get Autocloseable, we would not be able to use it along the auto closing capability offered through the the try block. I have made the change for extending Closeable class.

Reply:
why can we not use the auto closing capability ?

Reply:
My bad, but its possible, refactored.